test/screens/play/game_end_page_test.dart

```diff
@@ -0,0 +1,76 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:treasure_of_the_high_seas/model/game_result.dart';
+import 'package:treasure_of_the_high_seas/model/game_state_factory.dart';
+import 'package:treasure_of_the_high_seas/screens/main_menu/main_menu_page.dart';
+import 'package:treasure_of_the_high_seas/screens/play/game_end_page.dart';
+import 'package:treasure_of_the_high_seas/screens/play/play_page.dart';
+import 'package:treasure_of_the_high_seas/screens/play/player_hand.dart';
+
+import '../../test_utils.dart';
+
+void main() {
+  testWidgets('should display correct text for win and loss', (WidgetTester tester) async {
+    await tester.pumpWidget(createWidgetForTesting(child: GameEndPage(GameResult.WIN)));
+    expect(find.byType(GameEndPage), findsOneWidget);
+    expect(find.byType(PlayerHand), findsNothing);
+
+    var gameWinTextFinder = find.text(GameResult.WIN.description);
+    var gameLoseTextFinder = find.text(GameResult.LOSE.description);
+    expect(gameWinTextFinder, findsOneWidget);
+    expect(gameLoseTextFinder, findsNothing);
+
+    await tester.pumpWidget(createWidgetForTesting(child: GameEndPage(GameResult.LOSE)));
+    expect(find.byType(GameEndPage), findsOneWidget);
+    expect(find.byType(PlayerHand), findsNothing);
+
+    gameWinTextFinder = find.text(GameResult.WIN.description);
+    gameLoseTextFinder = find.text(GameResult.LOSE.description);
+    expect(gameWinTextFinder, findsNothing);
+    expect(gameLoseTextFinder, findsOneWidget);
+  });
+
+  testWidgets('new game button should start a new game', (WidgetTester tester) async {
+    await tester.pumpWidget(createWidgetForTesting(child: GameEndPage(GameResult.WIN)));
+    expect(find.byType(GameEndPage), findsOneWidget);
+    expect(find.byType(PlayerHand), findsNothing);
```

Comment:
We can do away with these assertions since we've made the GameEndPage in isolation, so we're not running through any code path that might make a `PlayerHand`